flowmachine/flowmachine/core/server/query_schemas/location_introversion.py

```diff
@@ -0,0 +1,52 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marshmallow import Schema, fields, post_load
+from marshmallow.validate import OneOf, Length
+
+from flowmachine.features import LocationIntroversion
+from .base_exposed_query import BaseExposedQuery
+from .custom_fields import AggregationUnit
+
+__all__ = ["LocationIntroversionSchema", "LocationIntroversionExposed"]
+
+
+class LocationIntroversionSchema(Schema):
+
+    start_date = fields.Date(required=True)
+    end_date = fields.Date(required=True)
+    aggregation_unit = AggregationUnit()
+    direction = fields.String(
+        required=True, validate=OneOf(["in", "out", "both"], default="both")
```

Comment:
This should be `required=False`, since it's an optional argument:
```suggestion
        required=False, validate=OneOf(["in", "out", "both"], default="both")
```

Reply:
Also, it looks like you added `default` inside the brackets of the call to `OneOf` rather than outside, which will throw an error because `default` is an argument to `fields.String`.